File: src/test/kotlin/io/mp3editor/MP3EditorTest.java
package io.mp3editor;

import org.testng.annotations.BeforeSuite;
import org.testng.annotations.Test;

import java.nio.file.Path;
import java.nio.file.Paths;

import static org.testng.Assert.assertEquals;

@Test
public class MP3EditorTest {
    // TODO: 4/18/2016 IMPLEMENT TESTS

    private MP3Editor editor;

    @BeforeSuite
    public void init() {
        editor = new MP3Editor();
    }

    @Test
    public void testDeduplicateFileName() {
        String newFileName = "newFileName";
        String currentFileName = "fileName";

        String expectedFileName = String.format("%s (%s)", newFileName, currentFileName);

        String deduplicatedFileName = editor.deduplicateFileName(newFileName, currentFileName);

        assertEquals(deduplicatedFileName, expectedFileName);
    }

    @Test
    public void testMakeFileName() {
        String firstPart = "first";
        String secondPart = "second";
        String expectedFileName = String.format("%s - %s", firstPart, secondPart);

        String fileName = editor.makeFileName(firstPart, secondPart);

        assertEquals(fileName, expectedFileName);
    }

    @Test
    public void testPrepareTargetPath() {
        String genreDir = "genre";
        String fileName = "fileName";
        String extension = ".extension";

        Path expectedPath = Paths.get(genreDir, fileName + extension);

        Path path = editor.prepareTargetPath(genreDir, fileName, extension);

        assertEquals(path, expectedPath);
    }
}

File: src/test/kotlin/io/mp3editor/MP3EditorTest.kt
package io.mp3editor

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import java.nio.file.Paths

class MP3EditorTest {
    // TODO: 4/18/2016 IMPLEMENT TESTS
    private lateinit var editor: MP3Editor

    @BeforeEach
    fun init() {
        editor = MP3Editor()
    }

    @Test
    fun testDeduplicateFileName() {
        val newFileName = "newFileName"
        val currentFileName = "fileName"

        val expectedFileName = "$newFileName ($currentFileName)"

        val deduplicatedFileName = editor.deduplicateFileName(newFileName, currentFileName)

        assertEquals(expectedFileName, deduplicatedFileName)
    }

    @Test
    fun testMakeFileName() {
        val firstPart = "first"
        val secondPart = "second"
        val expectedFileName = String.format("%s - %s", firstPart, secondPart)

        val fileName = editor.makeFileName(firstPart, secondPart)

        assertEquals(expectedFileName, fileName)
    }

    @Test
    fun testPrepareTargetPath() {
        val genreDir = "genre"
        val fileName = "fileName"
        val extension = ".extension"

        val expectedPath = Paths.get(genreDir, fileName + extension)

        val path = editor.prepareTargetPath(genreDir, fileName, extension)

        assertEquals(expectedPath, path)
    }
}
